doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,216 @@
+    * [1.4 Warm Restart requirements ](#14-warm-restart-requirements)
+  * [2 Modules Design](#2-modules-design)
+    * [2.1 Config DB](#21-config-db)
+    * [2.2 App DB](#22-app-db)
+    * [2.3 Orchestration Agent](#23-orchestration-agent)
+    * [2.4 CLI](#24-cli)
+
+###### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/09/2021  |     Prince Sunny   | Initial version                   |
+
+# About this Manual
+This document provides general information about the Vxlan Overlay ECMP feature implementation in SONiC with BFD support. This is an extension to the existing VNET Vxlan support as defined in the [Vxlan HLD](https://github.com/Azure/SONiC/blob/master/doc/vxlan/Vxlan_hld.md)
+# Scope
+This document describes the high level design of the Overlay ECMP feature and associated BFD support. General BFD support and configurations are beyond the scope of this document. 
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+|                          |                                |
+|--------------------------|--------------------------------|
+| BFD                      | Bidirectional Forwarding       |
+| VNI                      | Vxlan Network Identifier       |
+| VTEP                     | Vxlan Tunnel End Point         |
+| VNet                     | Virtual Network                |
+
+# 1 Requirements Overview
+
+## 1.1 Usecase
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_UseCase.png)
+
+## 1.1 Functional requirements
+
+At a high level the following should be supported:
+
+- Configure ECMP with Tunnel Nexthops (IPv4 and IPv6)
+- Tunnel Endpoint monitoring via BFD
+- Add/Withdraw Nexthop based on Tunnel or Endpoint health
+
+## 1.2 CLI requirements
+- User should be able to show the BFD session
+- User should be able to show the Vnet routes
+
+## 1.3 Warm Restart requirements
+No special handling for Warm restart support.
+
+# 2 Modules Design
+
+The following are the schema changes. 
+
+## 2.1 Config DB
+
+Existing Vxlan and Vnet tables. 
+
+### 2.1.1 VXLAN Table
+```
+VXLAN_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}} (OPTIONAL)
+```
+### 2.1.2 VNET/Interface Table
+```
+VNET|{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vni": {{vni}} 
+    "scope": {{"default"}} (OPTIONAL)
+    "peer_list": {{vnet_name_list}} (OPTIONAL)
+```
+
+## 2.2 APP DB
+
+### VNET
+
+The following are the changes for Vnet Route table
+
+Existing:
+
+``` 
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}} 
+    "endpoint": {{ip_address}} 
+    "mac_address":{{mac_address}} (OPTIONAL) 
+    "vni": {{vni}}(OPTIONAL) 
+```
+
+Proposed:
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
```

Comment:
No option to configure BFD monitoring via config-db?

Reply:
Do we need to repeat the IP address again in endpoint_monitor? what if some tunnel endpoints don't need monitoring? can we just have true/false?

Reply:
It can be empty if some endpoints don't need monitoring. This is similar to current multipath routing schema